hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java

```diff
@@ -1583,6 +1583,16 @@
     "hbase.regionserver.slowlog.systable.enabled";
   public static final boolean DEFAULT_SLOW_LOG_SYS_TABLE_ENABLED_KEY = false;
 
+  /**
+   * Number of rows in a batch operation above which a warning will be logged.
+   */
+  public static final String BATCH_ROWS_THRESHOLD_NAME = "hbase.rpc.rows.warning.threshold";
+
+  /**
+   * Default value of {@link #BATCH_ROWS_THRESHOLD_NAME}
+   */
+  public static final int BATCH_ROWS_THRESHOLD_DEFAULT = 5000;
+
```

Comment:
Why the move? HConstants is often considered an anti-pattern; better to have constants beside where they are used?

Reply:
We want `RsRpcServices` as well as `ReplicationSink` both to use these constants. Would you still recommend having duplicate copies?

Reply:
Ok. Two very disparate contexts. This is probably only case for HConstants.